src/Nethermind/Nethermind.Db/DbOnTheRocks.cs

```diff
@@ -163,97 +166,123 @@ private DbOptions BuildOptions(IDbConfig dbConfig)
                 {
                     if (value == null)
                     {
-                        _db.Remove(key, null, _writeOptions);
+                        Db.Remove(key, null, WriteOptions);
                     }
                     else
                     {
-                        _db.Put(key, value, null, _writeOptions);
+                        Db.Put(key, value, null, WriteOptions);
                     }
                 }
             }
         }
 
+        public KeyValuePair<byte[],byte[]>[] this[byte[][] keys] => Db.MultiGet(keys);
+
         public Span<byte> GetSpan(byte[] key)
         {
             UpdateReadMetrics();
-            return _db.GetSpan(key);
+            return Db.GetSpan(key);
         }
 
         public void DangerousReleaseMemory(in Span<byte> span)
         {
-            _db.DangerousReleaseMemory(in span);
+            Db.DangerousReleaseMemory(in span);
         }
 
         public void Remove(byte[] key)
         {
-            _db.Remove(key, null, _writeOptions);
+            Db.Remove(key, null, WriteOptions);
+        }
+
+        public IEnumerable<byte[]> GetAll()
+        {
+            using Iterator iterator = Db.NewIterator();
+            return GetAllCore(iterator);
         }
 
-        public byte[][] GetAll()
+        internal IEnumerable<byte[]> GetAllCore(Iterator iterator)
         {
-            Iterator iterator = _db.NewIterator();
-            iterator = iterator.SeekToFirst();
-            var values = new List<byte[]>();
+            iterator.SeekToFirst();
             while (iterator.Valid())
             {
-                values.Add(iterator.Value());
-                iterator = iterator.Next();
+                yield return iterator.Value();
+                iterator.Next();
             }
-
-            iterator.Dispose();
-
-            return values.ToArray();
         }
 
-        private byte[] _keyExistsBuffer = new byte[1];
         private ILogger _logger;
 
         public bool KeyExists(byte[] key)
         {
             // seems it has no performance impact
-            return _db.Get(key) != null;
+            return Db.Get(key) != null;
 //            return _db.Get(key, 32, _keyExistsBuffer, 0, 0, null, null) != -1;
         }
 
         public IDb Innermost => this;
 
         public void StartBatch()
         {
-            _currentBatch = new WriteBatch();
+            CurrentBatch = new WriteBatch();
         }
 
         public void CommitBatch()
         {
-            _db.Write(_currentBatch, _writeOptions);
-            _currentBatch.Dispose();
-            _currentBatch = null;
+            Db.Write(CurrentBatch, WriteOptions);
+            CurrentBatch.Dispose();
+            CurrentBatch = null;
         }
 
         public void Dispose()
         {
-            _db?.Dispose();
-            _currentBatch?.Dispose();
+            Db?.Dispose();
+            CurrentBatch?.Dispose();
+        }
+
+        public void Flush()
+        {
```

Comment:
what does it do?
can we  merge it with commitbatch?

Reply:
Flushes memory stuff to disk. I would leave it here. The idea was to flush at the end of Db migration. But as moved to flat files it is not used now.

Reply:
then let us remove, already trying to keep the IDb clean

Reply:
then let us remove, already trying to keep the IDb clean -unless it is just on rocks